### ISSUE TITLE
[Bug]: Phone number widget resets binding to static value for country code

### ISSUE THREAD

**Issue by jsartisan** [appsmithorg/appsmith#12705]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Whenever there is a binding for country code, the binding gets converted to static value on re-redenring of the widget.

### Steps To Reproduce

1. Drop phone number widget
2. Add `{{ "+91" }}` binding in country code
3. Re-arrange the position of the widget
4. Notice the value in country code in the property page. The binding gets converted to static value.the 

### Version

1.6.19

**Comment by aswathkk** [comment]
Another related issue:
https://www.loom.com/share/344f62a76bff459e87cea28b86aadba4

**Comment by dilippitchika** [comment]
Bumping this to critical back again as app viewer is getting impacted as per the video above cc @somangshu @sbalaji1192

**Comment by btsgh** [comment]
1. In Release URL - Created a Select Widget and a Phone number widget 
2. Populated Select Widget from a Excel datasource with Country Codes (value being phone code) 
3. Bound the DefaultCountryCode of the Phone Input widget to the data from Select Widget.
4. On Deploy, as we changed the dropdown options from the Select widget, the default country code of the phone widget changed accordingly.  
5. Also on the canvas, moving around the Phone Input widget as well as the Select widget did not impact any of the bindings. 
6. Checked the same scenario as above for Currency widget too. Looks to be working fine. 
7. Also tested the above two scenarios on Prod - looks to be working fine.
